code/modules/events/wizard/departmentrevolt.dm

```diff
@@ -1,56 +1,116 @@
+
+	var/announcement = event_control.announce
+	var/dangerous = event_control.dangerous_nation
+	var/department
+	if(event_control.picked_department)
+		department = event_control.picked_department
+		event_control.picked_department = null
+	else
+		department = pick(list("Uprising of Assistants", "Medical", "Engineering", "Science", "Supply", "Service", "Security") - cannot_pick)
+		if(!department)
+			message_admins("Department Revolt could not create a nation, as all the departments are independent! You have created nations, you madman!")
+	var/list/jobs_to_revolt	= list()
 	var/nation_name
-	var/list/citizens	=		list()
+	var/list/citizens = list()
 
-	tidecolor = pick("grey", "white", "yellow", "purple", "brown", "whatevercolorrepresentstheservicepeople")
-	switch(tidecolor)
-		if("grey") //God help you
+	switch(department)
+		if("Uprising of Assistants") //God help you
 			jobs_to_revolt = list("Assistant")
 			nation_name = pick("Assa", "Mainte", "Tunnel", "Gris", "Grey", "Liath", "Grigio", "Ass", "Assi")
-		if("white")
+		if("Medical")
 			jobs_to_revolt = GLOB.medical_positions
 			nation_name = pick("Mede", "Healtha", "Recova", "Chemi", "Viro", "Psych")
-		if("yellow")
+		if("Engineering")
 			jobs_to_revolt = GLOB.engineering_positions
 			nation_name = pick("Atomo", "Engino", "Power", "Teleco")
-		if("purple")
+		if("Science")
 			jobs_to_revolt = GLOB.science_positions
-			nation_name = pick("Sci", "Griffa", "Geneti", "Explosi", "Mecha", "Xeno")
-		if("brown")
+			nation_name = pick("Sci", "Griffa", "Geneti", "Explosi", "Mecha", "Xeno", "Nani", "Cyto")
+		if("Supply")
 			jobs_to_revolt = GLOB.supply_positions
 			nation_name = pick("Cargo", "Guna", "Suppli", "Mule", "Crate", "Ore", "Mini", "Shaf")
-		if("whatevercolorrepresentstheservicepeople") //the few, the proud, the technically aligned
+		if("Service") //the few, the proud, the technically aligned
 			jobs_to_revolt = GLOB.service_positions.Copy() - list("Assistant", "Prisoner")
 			nation_name = pick("Honka", "Boozo", "Fatu", "Danka", "Mimi", "Libra", "Jani", "Religi")
+		if("Security")
+			jobs_to_revolt = GLOB.security_positions
+			nation_name = pick("Securi", "Beepski", "Shitcuri", "Red", "Stunba", "Flashbango", "Flasha", "Stanfordi")
 
 	nation_name += pick("stan", "topia", "land", "nia", "ca", "tova", "dor", "ador", "tia", "sia", "ano", "tica", "tide", "cis", "marea", "co", "taoide", "slavia", "stotzka")
+	if(department == "Uprising of Assistants")
+		var/prefix = pick("roving clans", "barbaric tribes", "tides", "bandit kingdom", "tribal society", "marauder clans", "horde")
+		nation_name = "The [prefix] of [nation_name]"
 
-	var/datum/team/nation/nation = new
+	var/datum/team/nation/nation = new(null, jobs_to_revolt, department)
 	nation.name = nation_name
+	var/datum/team/department_target //dodges unfortunate runtime
+	if(!independent_departments.len)
+		department_target = null
+	else
+		department_target = pick(independent_departments)
```

Comment:
Inverse this. You should not be doing `if...else` with the original condition being a negative.